components/org.wso2.carbon.identity.api.user.recovery/src/test/java/org/wso2/carbon/identity/recovery/endpoint/impl/CaptchaApiServiceImplTest.java

```diff
@@ -61,8 +64,51 @@ public void tearDown() {
         mockedRecoveryUtil.close();
     }
 
-    @Test(description = "This method test, getReCaptcha method for username recovery")
-    public void testGetCaptcha() throws IOException {
+    @DataProvider(name = "captchaTestDataProvider")
+    public static Object[][] getCaptchaTestDataProvider() {
+
+        String reCaptchaEnterprise = CaptchaConstants.RE_CAPTCHA_TYPE_ENTERPRISE;
+
+        return new Object[][]{
+                {false, "", ""},
+                {true, "", "https://www.google.com/recaptcha/api.js"},
+                {true, reCaptchaEnterprise, "https://www.google.com/recaptcha/enterprise.js"}
+        };
+    }
+
+    @Test(description = "This method test, getReCaptcha method for username recovery",
+            dataProvider = "captchaTestDataProvider")
+    public void testGetCaptcha(boolean reCaptchaEnabled, String reCaptchaType,
+                               String reCaptchaAPI) throws IOException {
+
+        Properties sampleProperties = getSampleConfigFile();
+
+        sampleProperties.setProperty(CaptchaConstants.RE_CAPTCHA_ENABLED, String.valueOf(reCaptchaEnabled));
+        sampleProperties.setProperty(CaptchaConstants.RE_CAPTCHA_TYPE, reCaptchaType);
+        sampleProperties.setProperty(CaptchaConstants.RE_CAPTCHA_API_URL, reCaptchaAPI);
+
+        mockedRecoveryUtil.when(RecoveryUtil::getValidatedCaptchaConfigs).thenReturn(sampleProperties);
+        mockedRecoveryUtil.when(() -> RecoveryUtil.checkCaptchaEnabledResidentIdpConfiguration(Mockito.anyString(),
+                Mockito.anyString())).thenReturn(true);
+        Response response = captchaApiService.getCaptcha("ReCaptcha", "username-recovery",
+                "test.org");
+
+        ReCaptchaPropertiesDTO reCaptchaPropertiesDTO = response.readEntity(ReCaptchaPropertiesDTO.class);
+
+        assertEquals(reCaptchaPropertiesDTO.getReCaptchaEnabled().booleanValue(), reCaptchaEnabled);
+        if (reCaptchaPropertiesDTO.getReCaptchaType() == null) {
+            assertNull(reCaptchaPropertiesDTO.getReCaptchaType());
+        } else {
+            assertEquals(reCaptchaPropertiesDTO.getReCaptchaType(), reCaptchaType);
+        }
+        if (StringUtils.isBlank(reCaptchaAPI)) {
+            assertNull(reCaptchaPropertiesDTO.getReCaptchaAPI());
+        } else {
+            assertEquals(reCaptchaPropertiesDTO.getReCaptchaAPI(), reCaptchaAPI);
+        }
+    }
+
+    public Properties getSampleConfigFile() throws IOException {
```

Comment:
Do we need this a public method?

Reply:
It should be a private method. Will update.

Reply:
Update PR sent: https://github.com/wso2-extensions/identity-governance/pull/809

Reply:
Update not done due to https://github.com/wso2-extensions/identity-governance/pull/809#discussion_r1512094776